### PR TITLE
compute scuttle id for string shard keys

### PR DESCRIPTION
**Pull request by rsh-yam**
Fix for Issue #320. Also handles string shard keys for OCC_SHARD_KEY.

### CODE REVIEW

**Review by shtiencode — APPROVED** [review]
LGTM

File: worker/cppworker/worker/OCCChild.cpp [hunk 1]
```diff
@@ -5704,6 +5718,21 @@ uint OCCChild::compute_scuttle_id(unsigned long long _shardkey_val)
 	return scuttle_id;
 }
 
+uint OCCChild::compute_scuttle_id(std::string _shardkey_str_val)
+{
+	uint scuttle_id = 0;
+	switch(m_sharding_algo)
+	{
+		case MOD_ONLY:
```
Comment (venkatsridhar95) [inline worker/cppworker/worker/OCCChild.cpp:5726]: MOD_ONLY for string shard key isn't correct. If we enable any use cases later, we will run into the same issue of scuttle mismatch. We could either correct it or not support MOD_ONLY for now.

File: worker/cppworker/worker/OCCChild.cpp [hunk 2]
```diff
@@ -798,9 +799,16 @@ int OCCChild::handle_command(const int _cmd, std::string &_line)
 			uint32_t scuttle_id_val = 0;
 			if (values.size() && strcasecmp(name.c_str(), m_shard_key_name.c_str()) == 0)
 			{
-				// cast to long long
-				unsigned long long shard_val = StringUtil::to_ullong(values[0]);
```
[the remark this answers is not in the document]
Comment (rsh-yam) [reply worker/cppworker/worker/OCCChild.cpp:802]: There hasn't been any change made to the existing logic of using StringUtil::to_ullong to convert the numeric shard keys before computing scuttle id. The only change is detecting string shard keys and passing them directly to the hash function.

File: worker/cppworker/worker/OCCChild.cpp (hunk 1 above)
Comment (venkatsridhar95) [reply worker/cppworker/worker/OCCChild.cpp:5726]: The implementation matches with what we have in mux. 
https://github.com/paypal/hera/blob/master/lib/coordinatorsharding.go#L86

**Review by venkatsridhar95 — APPROVED** [review]

File: worker/cppworker/worker/OCCChild.cpp (hunk 1 above)
Comment (rsh-yam) [reply worker/cppworker/worker/OCCChild.cpp:5728]: added a comment for clarity as suggested

**Review by venkatsridhar95 — APPROVED** [review]

**Pull request merged**